flink-formats/flink-avro-glue-schema-registry/src/main/java/org/apache/flink/formats/avro/glue/schema/registry/GlueSchemaRegistryAvroSchemaCoderProvider.java

```diff
@@ -50,4 +51,29 @@ public GlueSchemaRegistryAvroSchemaCoderProvider(Map<String, Object> configs) {
     public GlueSchemaRegistryAvroSchemaCoder get() {
         return new GlueSchemaRegistryAvroSchemaCoder(transportName, configs);
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        GlueSchemaRegistryAvroSchemaCoderProvider that = (GlueSchemaRegistryAvroSchemaCoderProvider) o;
+        if (transportName == null)  {
+            if (that.transportName != null) {
+                return false;
+            }
+        } else if (!transportName.equals(that.transportName)) {
+            return false;
+        }
+        return configs.equals(that.configs);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(transportName, configs);
+    }
+
```

Comment:
Please confirm why you added these, and that it is acceptable per the coding guidelines:
- https://flink.apache.org/contributing/code-style-and-quality-java.html#equals--hashcode

Reply:
The test code relies on the existing implementations of [RegistryAvroSerializationSchema.equals](https://github.com/apache/flink/blob/master/flink-formats/flink-avro/src/main/java/org/apache/flink/formats/avro/RegistryAvroSerializationSchema.java#L115) and [RegistryAvroDeserializationSchema.equals](https://github.com/apache/flink/blob/master/flink-formats/flink-avro/src/main/java/org/apache/flink/formats/avro/RegistryAvroDeserializationSchema.java#L93), both of which then rely on their respective `schemaCoderProvider` having a valid `equals` implementation.  Although my only use case is testing, removing this function would break the equals contract for those containing classes.  

Because the class is effectively immutable, I don't believe it contravenes the guidelines.